src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,750 @@
+      props.onDrawerSlide && runOnJS(props.onDrawerSlide)(newOpenValue);
+
+      return newOpenValue;
+    });
+
+    const isDrawerOpen = useSharedValue(false);
+
+    const handleContainerLayout = ({ nativeEvent }: LayoutChangeEvent) => {
+      setContainerWidth(nativeEvent.layout.width);
+    };
+
+    const emitStateChanged = useCallback(
+      (newState: DrawerState, drawerWillShow: boolean) => {
+        'worklet';
+        props.onDrawerStateChanged &&
+          runOnJS(props.onDrawerStateChanged)?.(newState, drawerWillShow);
+      },
+      [props.onDrawerStateChanged]
+    );
+
+    const drawerAnimatedProps = useAnimatedProps(() => ({
+      accessibilityViewIsModal: isDrawerOpen.value,
+    }));
+
+    const overlayAnimatedProps = useAnimatedProps(() => ({
+      pointerEvents: isDrawerOpen.value ? ('auto' as const) : ('none' as const),
+    }));
+
+    // While the drawer is hidden, it's hitSlop overflows onto the main view by edgeWidth
+    // This way it can be swiped open even when it's hidden
+    const [edgeHitSlop, setEdgeHitSlop] = useState<HitSlop>(
+      isFromLeft
+        ? { left: 0, width: edgeWidth }
+        : { right: 0, width: edgeWidth }
+    );
+
+    // gestureOrientation is 1 if the gesture is expected to move from left to right and -1 otherwise
+    const gestureOrientation = useMemo(
+      () => sideCorrection * (drawerOpened ? -1 : 1),
+      [sideCorrection, drawerOpened]
+    );
+
+    const animateDrawer = useCallback(
+      (toValue: number, initialVelocity: number, animationSpeed?: number) => {
+        'worklet';
+        const willShow = toValue !== 0;
+        isDrawerOpen.value = willShow;
+        runOnJS(setEdgeHitSlop)(
+          isFromLeft
+            ? { left: 0, width: willShow ? undefined : edgeWidth }
+            : { right: 0, width: willShow ? undefined : edgeWidth }
+        );
+
+        emitStateChanged(DrawerState.SETTLING, willShow);
+        runOnJS(setDrawerState)(DrawerState.SETTLING);
+
+        if (props.hideStatusBar) {
+          StatusBar.setHidden(willShow, statusBarAnimation);
+        }
+
+        drawerTranslation.value = withSpring(
+          toValue,
+          {
+            // Velocity threshold does not matter as long as the destination is reached
+            // This prevents rubberbanding
+            restDisplacementThreshold: 1,
+            restSpeedThreshold: 10000,
+            overshootClamping: true,
+
+            velocity: initialVelocity,
+            mass: animationSpeed ? 1 / animationSpeed : 1,
+            damping: 50,
+            stiffness: 300,
+          },
+          (finished) => {
+            if (finished) {
+              emitStateChanged(DrawerState.IDLE, willShow);
+              runOnJS(setDrawerOpened)(willShow);
+              runOnJS(setDrawerState)(DrawerState.IDLE);
+              if (willShow) {
+                dragX.value = drawerWidth * sideCorrection;
+                props.onDrawerOpen && runOnJS(props.onDrawerOpen)?.();
```

Comment:
I'm unsure if we should access the entire `props` object on the UI thread instead of just the things that are needed.

Reply:
Fixed in [f14eff1](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/f14eff181dc67ff884a25922f6e4b10cd2d7da05)